File: Script/Fattoriale_iterativo/Fattoriale.py
import sys

def fattoriale_iter(n: int) -> int:
    risultato = 1
    for i in range(1, n + 1):
        risultato *= i
    return risultato

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print(f"Uso: python {sys.argv[0]} <numero>")
        sys.exit(1)

    n = int(sys.argv[1])
    print(f"Fattoriale iterativo({n}) = {fattoriale_iter(n)}")

File: Script/Fattoriale_iterativo/Fattoriale.c
#include <stdio.h>
#include <stdlib.h>

long fattoriale_iter(int n) {
    long risultato = 1;
    for (int i = 1; i <= n; i++) {
        risultato *= i;
    }
    return risultato;
}

int main(int argc, char *argv[]) {
    if (argc != 2) {
        printf("Uso: %s <numero>\n", argv[0]);
        return 1;
    }
    int n = atoi(argv[1]);
    printf("Fattoriale(%d) = %ld\n", n, fattoriale_iter(n));
    return 0;
}


